uk.co.compucorp.civicrm.hrleaveandabsences/tests/phpunit/CRM/HRLeaveAndAbsences/BAO/LeaveRequestDateTest.php

```diff
@@ -41,48 +41,42 @@ public function testALeaveRequestCantHaveMoreThanOneLeaveRequestDateWithTheSameD
   }
 
   public function testGetDatesForLeaveRequestReturnsTheDatesForASpecificLeaveRequest() {
-    LeaveRequestDate::create([
-      'date' => date('YmdHis'),
-      'leave_request_id' => 1,
-    ]);
-
-    LeaveRequestDate::create([
-      'date' => date('YmdHis', strtotime('+1 day')),
-      'leave_request_id' => 1,
-    ]);
-
-    LeaveRequestDate::create([
-      'date' => date('YmdHis', strtotime('+1 day')),
-      'leave_request_id' => 2,
-    ]);
-
-    LeaveRequestDate::create([
-      'date' => date('YmdHis', strtotime('+2 days')),
-      'leave_request_id' => 2,
+    $leaveRequest1 = LeaveRequestFabricator::fabricateWithoutValidation([
```

Comment:
The fact that this test broke with the changes is an indicator that `getDatesForLeaveRequest` should not have changed. If you compare the previous version of it with the new one, you can clearly see that now the method logic depends on details specific to the LeaveRequest (the reason why you now need the LeaveRequest Fabricator, where before all you needed was dates).

Also, if we think about this as an hierarchy, we can say that LeaveRequest is the parent and LeaveRequestDate is the child. Usually, in situations like this, it's ok for the parent to know details about the child, but it's not ok for the child to know details about the LeaveRequest (in this case, to know that it's parent has been deleted).